### PR TITLE
Release v0.49.0

### PR DESCRIPTION
**Pull request by seanchen1991**

### CODE REVIEW

**Review by Farhad-Shabani — COMMENTED** [review]
LGTM. Thank you!
Just per our discussion, let's move away from the `release.sh` and update the CI's release workflow to use the `cargo release ...` instead.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,70 @@
...
+  including `std` and `schema` features for `ibc-core`.
+  ([\#992](https://github.com/cosmos/ibc-rs/pull/992))
+- `[ibc-primitives]` streamline `Msg` trait and rename to `ToProto`
+ ([#993](https://github.com/cosmos/ibc-rs/issues/993))
+- `[ibc-core-host-types]` Implement custom JSON and Borsh deserialization for `ChainId` ([#996](https://github.com/cosmos/ibc-rs/pull/1013))
+- `[ibc-core-client-types]` Add a convenient `Status::verify_is_active` method.
+  ([#1005](https://github.com/cosmos/ibc-rs/pull/1005))
+- `[ibc-primitives]` Derive `Hash` on `Timestamp` instead of explicit
+  implementation ([#1011](https://github.com/cosmos/ibc-rs/pull/1005))
+- `[ibc-derive]` Use global paths in generated code by macros to prevent
```
Comment (Farhad-Shabani) [inline CHANGELOG.md:65]: Let's bump the `ibc-derive` version too. We had a change there.

Comment (seanchen1991) [reply CHANGELOG.md:65]: [babb171](https://github.com/cosmos/ibc-rs/pull/1021/commits/babb171958d5f7178ff1a3d8be3b76799cef60a5)

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
+
   release:
+    if: github.ref_type == 'tag'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
       - name: Publish crates
-        run: ./scripts/release.sh
+        run: cargo release --workspace --exclude ibc-derive
```
Comment (seanchen1991) [inline .github/workflows/release.yaml]: Do we not need to pass the `--execute` flag here?

Comment (Farhad-Shabani) [reply .github/workflows/release.yaml]: Yep, was tinkering with commands in the `release.yaml`, wanted to make sure I wouldn't accidentally push to crates-io.

**Comment by Farhad-Shabani** [comment]
I've gone ahead and revised the release job, and it should be all set now. 
By the way, before we move on, could @rnbguy please take a look at it? Thanks!

File: .changelog/v0.49.0/summary.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+This release continues the trend of further decoupling dependencies between the different ibc-rs
+sub-crates and modules. 
+
+In particular, the `prost` dependency is now only imported in the `ibc-primitives`
+crate; note that error variants originating from `prost` have largely been removed, which is a breaking
+change. The `bytes` dependency was also removed. Additionally, `CommitmentProofBytes` can now be 
+accessed without explicit ownership of the object for which the proof is being queried for.
```
Comment (rnbguy) [inline .changelog/v0.49.0/summary.md]:
```suggestion
accessed without explicit ownership of the object which the proof is being queried for. 
```

use `for which` or `which .. for`.

File: .changelog/v0.49.0/summary.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+This release continues the trend of further decoupling dependencies between the different ibc-rs
+sub-crates and modules. 
+
+In particular, the `prost` dependency is now only imported in the `ibc-primitives`
+crate; note that error variants originating from `prost` have largely been removed, which is a breaking
+change. The `bytes` dependency was also removed. Additionally, `CommitmentProofBytes` can now be 
+accessed without explicit ownership of the object for which the proof is being queried for. 
+
+Some other improvements of note include making the CosmWasm check more rigorous, streamlining the
```
Comment (rnbguy) [inline .changelog/v0.49.0/summary.md]:
```suggestion
Some other notable improvements include making the CosmWasm check more rigorous, streamlining the 
```

File: .changelog/v0.49.0/summary.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+This release continues the trend of further decoupling dependencies between the different ibc-rs
```
Comment (rnbguy) [inline .changelog/v0.49.0/summary.md]:
```suggestion
This release continues the trend of further decoupling dependencies between the different `ibc-rs`
```

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -1,27 +1,64 @@
 # Attempts to perform a release when a particular tag is pushed. This uses the
-# `./scripts/release.sh`, and assumes that the CRATES_TOKEN secret has been set
-# and contains an API token with which we can publish our crates to crates.io.
+# `cargo release`, and assumes that the CRATES_TOKEN secret has been set and
```
Comment (rnbguy) [inline .github/workflows/release.yaml]:
```suggestion
# `cargo release`, and assumes that the `CRATES_TOKEN` secret has been set and
```

File: .github/workflows/release.yaml [hunk 7]
```diff
@@ -1,27 +1,64 @@
...
 # crate being published depends on the crate published just prior, but the prior
 # crate isn't yet available via crates.io), one can simply rerun this workflow.
-# The release.sh script aims to be an idempotent operation, skipping the
-# publishing of crates that have already been published.
 name: Release
 
 on:
   push:
+    branches:
+      - "release/*"
```
Comment (rnbguy) [inline .github/workflows/release.yaml:17]: Why do we want to run it on `release/*` branches when we already run it on tags?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,71 @@
 # CHANGELOG
 
+## v0.49.0
+
+*January 2, 2024*
```
Comment (rnbguy) [inline CHANGELOG.md]:
```suggestion
*January 3, 2024*
```

File: .github/workflows/release.yaml (hunk 7 above)
Comment (rnbguy) [reply .github/workflows/release.yaml:17]: ah. `publish-check` runs on `release/` and `publish` and `gh-release` run on tags.

**Review by rnbguy — APPROVED** [review]
Added some comments. Otherwise, LGTM :rocket:

File: CHANGELOG.md (hunk 8 above)
Comment (seanchen1991) [reply CHANGELOG.md]: [9239ca7](https://github.com/cosmos/ibc-rs/pull/1021/commits/9239ca79b6c35c5dfa56237c87ca45bea7f19120)

File: .github/workflows/release.yaml (hunk 6 above)
Comment (seanchen1991) [reply .github/workflows/release.yaml]: [9239ca7](https://github.com/cosmos/ibc-rs/pull/1021/commits/9239ca79b6c35c5dfa56237c87ca45bea7f19120)

File: .changelog/v0.49.0/summary.md (hunk 5 above)
Comment (seanchen1991) [reply .changelog/v0.49.0/summary.md]: [9239ca7](https://github.com/cosmos/ibc-rs/pull/1021/commits/9239ca79b6c35c5dfa56237c87ca45bea7f19120)

File: .changelog/v0.49.0/summary.md (hunk 4 above)
Comment (seanchen1991) [reply .changelog/v0.49.0/summary.md]: [9239ca7](https://github.com/cosmos/ibc-rs/pull/1021/commits/9239ca79b6c35c5dfa56237c87ca45bea7f19120)

File: .changelog/v0.49.0/summary.md (hunk 3 above)
Comment (seanchen1991) [reply .changelog/v0.49.0/summary.md]: [9239ca7](https://github.com/cosmos/ibc-rs/pull/1021/commits/9239ca79b6c35c5dfa56237c87ca45bea7f19120)

**Pull request merged**